node-liveness-checker/Dockerfile

```diff
@@ -0,0 +1,4 @@
+ARG base_image
+FROM ${base_image}
+# Needed by run custom build command for 'concordium-rust-sdk'.
+RUN rustup component add rustfmt
```

Comment:
Please do not pollute the top-level and put these build scripts into scripts.

Reply:
I am following the convention used in other public and private repositories, where the `Jenkinsfile` and `Dockerfile` also live in the root of the project. See https://github.com/Concordium/concordium-rosetta, https://github.com/Concordium/concordium-wallet-proxy and https://github.com/Concordium/concordium-ledger-app

Reply:
I think `concordium-base` and `concordium-node` are the only repos where the docker/jenkinsfiles don't live in the source root. I prefer them to be there. This is the common place for Dockerfile (and I think for Jenkinsfile too):

https://docs.docker.com/engine/reference/builder/:

> Traditionally, the Dockerfile is called Dockerfile and located in the root of the context [...]

Reply:
This is just the lazy default with magical behaviour that makes it harder to understand for newcomers. 

I am strongly against this pollution. And there should also be an inventory of scripts, i.e., a README that details their intended usage. 

If you want to have scripts in the repository they should conform to the same standard of documentation as source code.

Reply:
I think not following common conventions is what's most prone to confusion.

Reply:
Having `Dockerfile`, `Jenkinsfile` and other files like `Cargo.toml` and `Cargo.lock` in the project root is considered industry standard and is expected by newcomers, therefore not causing any confusion. It is what anyone who has worked with Docker or Jenkins in the past would expect.